### ISSUE TITLE
0.18.0 release

### ISSUE THREAD

**Issue by QuLogic** [SciTools/cartopy#1458]
There are currently 20 open issues/PRs on the [0.18 milestone](https://github.com/SciTools/cartopy/milestone/25). 8 are PRs and mostly bug fixes. I have just tagged 0.18.0b1 and it looks like #1161 is working because it was automatically pushed to PyPI.

There's one major thing I'd like to merge for 0.18 (#1422), and the rest of the PRs are bug fixes which should be okay. The issues are either fixed by them, or can wait.

However, I tagged a beta, not release candidate, because there are some major changes to labelling, and I would like to know what effect, if any, this might have on downstream projects. Let me ping the ones I know of:
* geoplot @ResidentMario
* yt @munkm @matthewturk (you're the last two people to touch Cartopy stuff, hopefully the right people to ping)
* iris @SciTools/developers 

Please test with the 0.18.0 beta release and open issues if you find any.

**Comment by ResidentMario** [comment]
geoplot doesn't have any labeling features (yet) so no problems here.

**Comment by QuLogic** [comment]
I should also mention #1213 (a corollary to #1422) which changes spines a bit, and may have an effect on any image tests, if you have any.

**Comment by xylar** [comment]
I'm testing a build of 0.18.0b1 on conda-forge as you can see above.  I'm running into trouble with:
https://github.com/SciTools/cartopy/blob/master/setup.py#L244-L247
It seems that proj >=6.2 doesn't require this special treatment (the library is just `proj.lib`, though there is a copy as `proj_6_2.lib`) and proj >=6.3 seems to have dropped the numbering entirely (there is just `proj.lib`).  At least this is the case for the packages on `conda-forge`.

I'm happy to create a separate issue for this if you prefer.

**Comment by QuLogic** [comment]
Yes, I do remember something about the name changing on Windows to be more consistent.

That reminds me, I should ping packagers as well. I'm not sure who does anything except Fedora though (i.e., me). I think for Debian it may be @sebastic.

**Comment by sebastic** [comment]
> I think for Debian it may be @sebastic.

It's also @avalentino.

The Debian package build failed due to a test failure because there is no network in the build environment.

The patch is forwarded in #1461, but that won't be merged because I refuse to sign CLAs out of principle.

**Comment by xylar** [comment]
Beta package can now be installed from conda-forge with:
```
conda install -c conda-forge/label/cartopy_dev cartopy
```

**Comment by stefraynaud** [comment]
The list of contributors should probably updated before the final release.

**Comment by QuLogic** [comment]
I have not touched docs yet at all; there are many things to do still.

**Comment by QuLogic** [comment]
@sebastic Fedora runs a [build on s390x](https://koji.fedoraproject.org/koji/taskinfo?taskID=41454024) without issue, so I'm not sure if it's Cartopy, or one of its dependencies. There is only one [patch for new FreeType](https://src.fedoraproject.org/rpms/python-cartopy/blob/master/f/0001-Increase-tolerance-for-new-FreeType.patch), but it should not affect those tests.

**Comment by sebastic** [comment]
> Fedora runs a [build on s390x](https://koji.fedoraproject.org/koji/taskinfo?taskID=41454024) without issue, so I'm not sure if it's Cartopy, or one of its dependencies.

It's likely GEOS, the Fedora build is still building with 3.7.1 instead of 3.8.0.

**Comment by greglucas** [comment]
@QuLogic, it looks like we are down to ~2-3 open PRs in that list now, all of which are in pretty good shape. I think you mentioned you were going to clean up some of my typos/update the docs. Let me know if you want me to take a look at anything else for the 0.18 release.

**Comment by QuLogic** [comment]
I just tagged v0.18.0b2, which should fixed the issues reported here. The other major change is that #1422 was merged, so the background patch will act a little different.

Please let us know if there are any remaining or new issues; I hope to tag a final in about 2 weeks.

**Comment by xylar** [comment]
For those interested in testing via a conda environment, I built the conda-forge version of `0.18.0b2`, which can be installed with:
```
conda config --set channel_priority strict
conda create -y -n test --override-channels -c conda-forge/label/cartopy_dev \
    -c conda-forge -c defaults python=3.7 cartopy=0.18.0b2
```
I wouldn't recommend testing the package by installing into an existing conda envrionment because there is too high a chance of getting incompatible packages from different channels and labels.  (This is sound advice in general for conda-forge, but particularly when using a `*_dev` label.)

**Comment by greglucas** [comment]
I'm not sure if you want to tag another beta or not now that we have a few other bug fixes in. Notably:
- Fixing clipping issues for inline labels
- Support for MPL 3.3

I'm in favor of just making the 0.18 release without another beta. Then we can try and target a 0.19 pretty soon eliminating Python 2.

**Comment by sebastic** [comment]
Why not an RC instead of beta, that's more common than going straight from beta to final.

**Comment by QuLogic** [comment]
I will tag an RC after #1536 is in, and then the final will be at the end of the month. That's not a long time, but the changes since the last beta have only been small bug fixes and cosmetics.

**Comment by QuLogic** [comment]
I tagged rc1 and made a branch for 0.18.

**Comment by stefraynaud** [comment]
It's becoming closer and closer to the release.
And how about the docs? 

I'm not alone to think that it should be clarified, without rewriting everything. See for instance #1510. The [xarray](http://xarray.pydata.org/en/stable/)  doc is complex and very clear, with a toc that is always visible and divided by toctree captions. Don't wait one more year for that!

I was ready to propose quick modifications just to start, but there are to mush sphinx hacks in the docs, like the sphinx gallery usage and extension.

**Comment by dopplershift** [comment]
👍 on enhancing docs. 👎 on anything that delays this release. Now, with any luck, we could move to more frequent releases, rather than less.

**Comment by stefraynaud** [comment]
Ok, it makes it clear now :)

**Comment by QuLogic** [comment]
0.18 is out now! 🎉
